iexec/src/main/scala/gridscale/iexec/package.scala

```diff
@@ -0,0 +1,83 @@
+package gridscale
+
+import effectaside._
+import gridscale.cluster.BatchScheduler.BatchJob
+import gridscale.cluster.{ BatchScheduler, HeadNode }
+import gridscale.tools._
+import monocle.macros._
+import squants._
+import squants.information._
+
+package object iexec {
+
+  case class IEXECJobDescription(
+    workDirectory: String,
+    IexecFilesPath: String,
+    dappAddress: String,
+    arguments: String,
+    dappCost: Int)
+
+  object impl {
+
+    def toIexec(description: IEXECJobDescription) =
+      s"""
+         |#!/bin/bash
+         |export PATH="${description.IexecFilesPath}:$${PATH}"
+         |cd ${description.workDirectory}
+         |iexec account login
+         |iexec account allow ${description.dappCost}
```

Comment:
we will want a mode where you allow more than the dApp price to be able to submit multiple jobs without having to wait for a transaction to be mined everytime, so this should be a separate function
you can verify the account balance before submission (`iexec account show`) and compare it to the dapp price

later with a finer integration you'll be able to query the dApp's price from its smart contract directly